spring-core/src/main/java/org/springframework/util/StopWatch.java

```diff
@@ -338,6 +376,34 @@ private static double nanosToSeconds(long duration) {
 		return duration / 1_000_000_000.0;
 	}
 
+	private static double convertTimeUnit(double amount, TimeUnit from, TimeUnit to) {
+		if (from.ordinal() < to.ordinal()) {
+			return amount / from.convert(1, to);
+		} else {
+			return amount * to.convert(1, from);
+		}
+	}
+
+	private static String getTimeUnitAbbreviation(TimeUnit timeUnit) {
```

Comment:
Why use name() string rather than enum for lookups?
What's the good of it?
eg:
```
private static String getTimeUnitAbbreviation(TimeUnit timeUnit) {
      switch (timeUnit) {
          case NANOSECONDS:
              return "ns";
          case MICROSECONDS:
              return "μs";
          case MILLISECONDS:
              return "ms";
          case SECONDS:
              return "sec";
          case MINUTES:
              return "min";
          case HOURS:
              return "hrs";
          case DAYS:
              return "days";
          default:
              throw new IllegalArgumentException("No TimeUnit equivalent for " + timeUnit);
      }
  }
```

Reply:
Updated